zone/raids.cpp

```diff
@@ -1486,70 +1605,84 @@ void Raid::SaveRaidMOTD()
-                                    (unsigned long)GetID());
-    auto results = database.QueryDatabase(query);
-    if (!results.Success())
-        return false;
-
-	if(results.RowCount() == 0) {
-		LogError("Error getting raid members for raid [{}]: [{}]", (unsigned long)GetID(), results.ErrorMessage().c_str());
-        disbandCheck = true;
-        return false;
-    }
-
-    int index = 0;
-    for(auto row = results.begin(); row != results.end(); ++row) {
-        if(!row[0])
-            continue;
-
-        members[index].member = nullptr;
-        strn0cpy(members[index].membername, row[0], 64);
-		uint32 groupNum = Strings::ToUnsignedInt(row[1]);
-        if(groupNum > 11)
-            members[index].GroupNumber = 0xFFFFFFFF;
-        else
-            members[index].GroupNumber = groupNum;
-
-        members[index]._class = Strings::ToInt(row[2]);
-        members[index].level = Strings::ToInt(row[3]);
-        members[index].IsGroupLeader = Strings::ToInt(row[4]);
-        members[index].IsRaidLeader = Strings::ToInt(row[5]);
-        members[index].IsLooter = Strings::ToInt(row[6]);
-        ++index;
-    }
+	const auto query = fmt::format(
+		"SELECT name, groupid, _class, level, "
+		"isgroupleader, israidleader, islooter, bot_id "
+		"FROM raid_members WHERE raidid = {} ORDER BY groupid",
+		GetID()
+	);
+
+	auto results = database.QueryDatabase(query);
+	if (!results.Success()) {
+		return false;
+	}
+
+	if (!results.RowCount()) {
+		LogError("Error getting raid members for raid [{}]: [{}]", GetID(), results.ErrorMessage());
+		disbandCheck = true;
+		return false;
+	}
+
+	int index = 0;
+	for (auto row: results) {
+		if (!row[0]) {
+			continue;
+		}
+
+		members[index].member = nullptr;
+		strn0cpy(members[index].membername, row[0], sizeof(members[index].membername));
+		uint32 group_id = Strings::ToUnsignedInt(row[1]);
+
+		if (group_id > 11) {
+			members[index].GroupNumber = RAID_GROUPLESS;
+		} else {
+			members[index].GroupNumber = group_id;
+		}
 
+		members[index]._class        = Strings::ToUnsignedInt(row[2]);
+		members[index].level         = Strings::ToUnsignedInt(row[3]);
+		members[index].IsGroupLeader = Strings::ToBool(row[4]);
+		members[index].IsRaidLeader  = Strings::ToBool(row[5]);
+		members[index].IsLooter      = Strings::ToBool(row[6]);
+		members[index].IsBot         = Strings::ToBool(row[7]) > 0;
+		++index;
+	}
+  
 	return true;
 }
 
 void Raid::VerifyRaid()
 {
-	for(int x = 0; x < MAX_RAID_MEMBERS; x++)
-	{
-		if(strlen(members[x].membername) == 0){
-			members[x].member = nullptr;
-		}
-		else{
-			Client *c = entity_list.GetClientByName(members[x].membername);
-			if(c){
-				members[x].member = c;
-			}
-			else{
-				members[x].member = nullptr;
+	for (auto& m : members) {
+		if(strlen(m.membername) == 0){
+			m.member = nullptr;
+		} else {
+			auto* c = entity_list.GetClientByName(m.membername);
+			auto* b = entity_list.GetBotByBotName(m.membername);
```

Comment:
this is gross and couples the logic with bots, we shouldn't be doing this if bots aren't enabled in core business logic